File: WoofAndroid/design-system/src/main/kotlin/com/perrystreet/woof/designsystem/atomic/atoms/tag/roles/TagStyleRole.kt
package com.perrystreet.woof.designsystem.atomic.atoms.tag.roles

import androidx.compose.runtime.Composable
import androidx.compose.ui.graphics.Color
import com.perrystreet.woof.designsystem.theme.Theme

enum class TagStyleRole {
    Neutral {
        @Composable
        override fun backgroundColor(): Color = Theme.colors.surfaceContainerHigh

        @Composable
        override fun textColor(): Color = Theme.colors.onSurface
    },
    OnScrim {
        @Composable
        override fun backgroundColor(): Color = Theme.colors.scrimContainer

        @Composable
        override fun textColor(): Color = Theme.colors.onScrim
    },
    Accent {
        @Composable
        override fun backgroundColor(): Color = Theme.colors.primary

        @Composable
        override fun textColor(): Color = Theme.colors.onPrimary
    },
    ;

    @Composable
    abstract fun backgroundColor(): Color

    @Composable
    abstract fun textColor(): Color
}

File: WoofIOS/SwiftPackages/DesignSystem/Sources/DesignSystem/Atomic/Atoms/Tag/Roles/TagStyleRole.swift
import SwiftUI

public enum TagStyleRole {
    case neutral
    case onScrim
    case accent

    public func backgroundColor(from theme: ThemeImplementing) -> Color {
        switch self {
        case .neutral: theme.colors.surfaceContainerHigh
        case .onScrim: theme.colors.scrimContainer
        case .accent: theme.colors.primary
        }
    }

    public func textColor(from theme: ThemeImplementing) -> Color {
        switch self {
        case .neutral: theme.colors.onSurface
        case .onScrim: theme.colors.onScrim
        case .accent: theme.colors.onPrimary
        }
    }
}
